src/org/zaproxy/zap/extension/ascan/ActiveScan.java

```diff
@@ -289,7 +296,27 @@ public void run() {
             });
         }
 	}
-	
+
+    private void removeHistoryReference(HistoryReference remRef) {
+        if (View.isInitialised()) {

```

Comment:
Since this affects the new methods I think it's related (I guess you could leave the add methods out, but it's just a deletion/tweak).